pkg/ovs/openflow/ofctrl_action.go

```diff
@@ -303,14 +337,35 @@ func (a *ofFlowAction) LoadRegMark(marks ...*RegMark) FlowBuilder {
 	return fb
 }
 
-// LoadToPktMarkRange is an action to load data into pkt_mark at specified range.
+// LoadPktMarkRange is an action to load data into pkt_mark at specified range.
 func (a *ofFlowAction) LoadPktMarkRange(value uint32, rng *Range) FlowBuilder {
-	return a.LoadRange(NxmFieldPktMark, uint64(value), rng)
+	pktMarkField, _ := openflow15.FindFieldHeaderByName(NxmFieldPktMark, true)
+	var valueBytes, maskBytes [4]byte
+	valueData := value
+	mask := uint32(0)
+	if rng != nil {
+		mask = ^mask >> (32 - rng.Length()) << rng.Offset()
+		binary.BigEndian.PutUint32(maskBytes[:], mask)
+		pktMarkField.Mask = util.NewBuffer(maskBytes[:])
+		valueData = valueData << rng.Offset()
+	}
+	binary.BigEndian.PutUint32(valueBytes[:], valueData)
```

Comment:
In libOpenflow, there is no existing exact Message type for `NXM_NX_PKT_MARK`, so we should use the general type Buffer. Then we need to have a byte slice to construct Buffer.

Reply:
I meant why we create new slices with `[:]` so many times, but now I see `valueBytes` and `maskBytes` are arrays.
Since the method mainly use slices, should it just declare slices in the beginning, instead of declaring arrays and keep creating slices from it?
```
valueBytes := make([]byte, 4)
maskBytes := make([]byte, 4)
```

Reply:
updated.